### PR TITLE
PLATFORM-2366: [PHP-FPM] Verify Apache-related PHP functions calls

### PR DESCRIPTION
**Pull request by macbre**
[PLATFORM-2366](https://wikia-inc.atlassian.net/browse/PLATFORM-2366)

As we're giving fpm a try let's rely on `$_SERVER` or MediaWiki's `RequestContext` instead of Apache functions.

``` php
$ ack-grep apache_ --php
includes/wikia/transaction/Transaction.php
274:        if ( function_exists( 'apache_response_headers' ) ) {
275:            $isCacheable = self::isCacheable( apache_response_headers() );

extensions/wikia/SkinChooser/SkinChooser.class.php
195:        if ( is_null( $useskin ) && function_exists( 'apache_request_headers' ) ) {
196:            $headers = apache_request_headers();

extensions/wikia/Tasks/proxy/proxy.php
6:$headers = apache_request_headers();
```

@artursitarski / @wladekb

### CODE REVIEW

**Comment by artursitarski** [comment]
:+1:

File: includes/wikia/transaction/Transaction.php [hunk 1]
```diff
@@ -237,15 +237,29 @@ public static function onAfterSmwfGetStore( $store ) {
...
+		/**
+		 * $headers will have entries like the following ones:
+		 *
+		 * 'X-Served-By: dev-macbre'
+		 * 'X-Trace-Id: 2d425caa-19a9-4f78-a5e7-8bdda3286f0d'
+		 *
+		 * Map it to an associative array
+		 */
+		$headersMap = [];
+		foreach($headers as $header) {
```
Comment (wladekb) [inline includes/wikia/transaction/Transaction.php]: we're actually using only the `Cache-Control`, the `if` could be embedded inside foreach and you could skip generating the entire `$headersMap`

File: includes/wikia/transaction/Transaction.php [hunk 2]
```diff
@@ -237,15 +237,29 @@ public static function onAfterSmwfGetStore( $store ) {
...
+			list($key, $val) = explode(':', $header, 2);
+			$headersMap[ strtoupper( $key ) ] = $val;
```
Comment (wladekb) [inline includes/wikia/transaction/Transaction.php]: I'd say it's more common to use `strtolower` in such cases but it's for sure arguable

**Comment by wladekb** [comment]
generally lgtm, I left a few comments to consider though

File: includes/wikia/transaction/Transaction.php (hunk 1 above)
Comment (macbre) [inline includes/wikia/transaction/Transaction.php]: Good point -  refactored in a24fec7

**Comment by wladekb** [comment]
lgtm

**Comment by macbre** [comment]
@artursitarski - backported to tomorrow's release branch in 92936071

**Comment by artursitarski** [comment]
Thank you!

**Pull request merged**